File: square/src/main/java/com/raxdenstudios/square/interceptor/ApplicationInterceptor.java
package com.raxdenstudios.square.interceptor;

import android.app.Application;
import android.content.res.Configuration;

import com.raxdenstudios.square.lifecycle.ApplicationLifecycle;

/**
 * Created by Ángel Gómez
 *
 * This abstract class defines the basis of an application interceptor.
 */
public abstract class ApplicationInterceptor<TCallback extends InterceptorCallback> extends BaseInterceptor<TCallback> implements ApplicationLifecycle {

    protected Application mApplication;

    public ApplicationInterceptor(Application application) {
        super(application);
        mApplication = application;
    }

    public ApplicationInterceptor(Application application, TCallback callback) {
        super(application, callback);
        mApplication = application;
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {

    }

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public void onLowMemory() {

    }

    @Override
    public void onTrimMemory(int level) {

    }

}

File: square/src/main/java/com/raxdenstudios/square/interceptor/ApplicationInterceptor.kt
package com.raxdenstudios.square.interceptor

import android.app.Application
import android.content.res.Configuration
import com.raxdenstudios.square.lifecycle.ApplicationLifecycle

/**
 * Created by Ángel Gómez
 *
 * This abstract class defines the basis of an application interceptor.
 */
abstract class ApplicationInterceptor<TCallback : InterceptorCallback>(
        protected var application: Application,
        callback: TCallback? = null)
    : BaseInterceptor<TCallback>(application, callback),
        ApplicationLifecycle {

    override fun onConfigurationChanged(newConfig: Configuration?) {}

    override fun onLowMemory() {}

    override fun onTrimMemory(level: Int) {}

}
